src/apps/tco2_dashboard/app.py

```diff
@@ -768,102 +953,267 @@ def get_daily_agg_klima_retirements():
+        df_bridged_mco2,
+        df_bridged_tx_mco2,
+        df_retired_mco2,
+        df_retired_mco2_info,
+        df_verra,
+        verra_fallback_note,
+        df_prices,
+        current_price_only_token_list,
+        price_source,
+        df_holdings,
+        polygon_carbon_metrics_df,
+        eth_carbon_metrics_df,
+        celo_carbon_metrics_df,
+    )
 
+
+@cache.memoize()
+def generate_layout():
+    curr_time_str = datetime.utcnow().strftime("%b %d %Y %H:%M:%S UTC")
+    curr_day_str = datetime.utcnow().strftime("%d")
+    try:
+        if curr_day_str in BACKUP_DATA_DAYS:
+            client = initialize_client_s3(
+                region_name=DO_SPACES_REGION, endpoint_url=DO_SPACES_ENDPOINT
+            )
+            print(
+                f"Starting to fetch data and simultaneously create Backup for Day : {curr_day_str}"
+            )
+            (
+                df,
+                df_retired,
+                df_deposited,
+                df_redeemed,
+                df_pool_retired,
+                df_bridged_mco2,
+                df_bridged_tx_mco2,
+                df_retired_mco2,
+                df_retired_mco2_info,
+                df_verra,
+                verra_fallback_note,
+                df_prices,
+                current_price_only_token_list,
+                price_source,
+                df_holdings,
+                polygon_carbon_metrics_df,
+                eth_carbon_metrics_df,
+                celo_carbon_metrics_df,
+            ) = run_get_data_functions(
+                use_backup=False,
+                client=client,
+                backup_day=curr_day_str,
+                backup_data=True,
+            )
+            update_backup_time(
+                client,
+                DO_SPACES_BUCKET_NAME,
+                filename="Exact_Backup_Time",
+                endpoint=DO_SPACES_ENDPOINT,
+                backup_day=curr_day_str,
+                backup_time=curr_time_str,
+            )
+            print("Backup creation successful")
+        else:
+            print("Starting to Fetch Data")
+            (
+                df,
+                df_retired,
+                df_deposited,
+                df_redeemed,
+                df_pool_retired,
+                df_bridged_mco2,
+                df_bridged_tx_mco2,
+                df_retired_mco2,
+                df_retired_mco2_info,
+                df_verra,
+                verra_fallback_note,
+                df_prices,
+                current_price_only_token_list,
+                price_source,
+                df_holdings,
+                polygon_carbon_metrics_df,
+                eth_carbon_metrics_df,
+                celo_carbon_metrics_df,
+            ) = run_get_data_functions(
+                use_backup=False,
+                client=None,
+                backup_day=curr_day_str,
+                backup_data=False,
+            )
+        print("Data Fetch Successful")
+    except Exception:
```

Comment:
This is too generic - we need to catch specific errors to avoid creating unpredictable and difficult to debug behavior